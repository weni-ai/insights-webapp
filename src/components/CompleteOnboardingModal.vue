<template>
  <UnnnicModalDialog
    :modelValue="props.showModal"
    class="finish-onboarding-modal"
    :primaryButtonProps="{ text: 'Ok' }"
    hideSecondaryButton
    showActionsDivider
    size="sm"
    @primary-button-click="emit('finish-onboarding')"
  >
    <section class="finish-onboarding-modal__content">
      <h1 class="finish-onboarding-modal__content__title">
        {{ $t('onboarding.complete_modal.title') }}
      </h1>
      <p class="finish-onboarding-modal__content__info">
        {{ $t('onboarding.complete_modal.text') }}
      </p>
    </section>
  </UnnnicModalDialog>
</template>

<script setup>
const props = defineProps({
  showModal: {
    type: Boolean,
    required: true,
  },
});

const emit = defineEmits(['close', 'finish-onboarding']);
</script>

<style lang="scss" scoped>
.finish-onboarding-modal {
  :deep(.unnnic-modal-dialog__container__actions) {
    display: flex;
    width: 100%;
    > .unnnic-modal-dialog__container__actions__primary-button {
      flex: 1;
    }
  }
  &__content {
    display: flex;
    flex-direction: column;
    align-items: center;
    text-align: center;
    &__title {
      font-size: $unnnic-font-size-title-sm;
      font-weight: $unnnic-font-weight-black;
      line-height: $unnnic-line-height-large * 1.75;
      color: $unnnic-color-neutral-darkest;
    }
    &__info {
      font-size: $unnnic-font-size-body-gt;
      font-weight: $unnnic-font-weight-regular;
      line-height: $unnnic-line-height-large * 1.375;
      color: $unnnic-color-neutral-cloudy;
      margin-top: $unnnic-spacing-xs;
    }
  }
}
</style>
