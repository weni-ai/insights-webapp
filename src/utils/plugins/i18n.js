/* eslint-disable camelcase */
import { createI18n } from 'vue-i18n';

import moment from 'moment';

import pt_br from '@/locales/pt_br.json';
import en from '@/locales/en.json';
import es from '@/locales/es.json';

import('moment/dist/locale/es.js');
import('moment/dist/locale/pt-br.js');

moment.locale('en');

const languages = {
  'pt-br': pt_br,
  en,
  es,
};

const messages = Object.assign(languages);

export default createI18n({
  locale: 'en',
  fallbackLocale: 'en',
  messages,
  silentTranslationWarn: true,
  silentFallbackWarn: true,
});
