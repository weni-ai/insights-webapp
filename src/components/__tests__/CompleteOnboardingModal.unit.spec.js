import { beforeEach, describe, expect, it } from 'vitest';
import { mount, config } from '@vue/test-utils';
import CompleteOnboardingModal from '@/components/CompleteOnboardingModal.vue';
import i18n from '@/utils/plugins/i18n';

beforeAll(() => {
  config.global.plugins = config.global.plugins.filter(
    (plugin) => plugin !== i18n,
  );
});

afterAll(() => {
  if (!config.global.plugins.includes(i18n)) {
    config.global.plugins.push(i18n);
  }
});

describe('CompleteOnboardingModal', () => {
  let wrapper;

  const createWrapper = (props) => {
    return mount(CompleteOnboardingModal, {
      props,
      global: {
        mocks: {
          $t: (key) => key,
        },
        stubs: {
          UnnnicModalDialog: {
            template: '<div><slot /><slot name="primary-button"></slot></div>',
          },
        },
      },
    });
  };

  beforeEach(() => {
    wrapper = createWrapper({ showModal: true });
  });

  describe('Props', () => {
    it('should receive showModal as true', () => {
      expect(wrapper.props('showModal')).toBe(true);
    });

    it('should pass showModal prop to UnnnicModalDialog', () => {
      const modalDialog = wrapper.findComponent(
        '[data-test-id="modal-dialog"]',
      );

      expect(modalDialog.attributes('modelvalue')).toBe('true');
    });
  });

  describe('Rendering Content', () => {
    it('should display the modal title from translation', () => {
      const title = wrapper.find('.finish-onboarding-modal__content__title');
      expect(title.exists()).toBe(true);
      expect(title.text()).toBe(
        wrapper.vm.$t('onboarding.complete_modal.title'),
      );
    });

    it('should display the modal info text from translation', () => {
      const info = wrapper.find('.finish-onboarding-modal__content__info');
      expect(info.exists()).toBe(true);
      expect(info.text()).toBe(wrapper.vm.$t('onboarding.complete_modal.text'));
    });
  });

  describe('Primary Button Click', () => {
    it('should emit "finish-onboarding" when primary button is clicked', async () => {
      const primaryButton = wrapper.findComponent(
        '[data-test-id="modal-dialog"]',
      );

      await primaryButton.vm.$emit('primary-button-click');
      expect(wrapper.emitted('finish-onboarding')).toBeTruthy();
    });
  });

  describe('Snapshot', () => {
    it('should match the snapshot', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
