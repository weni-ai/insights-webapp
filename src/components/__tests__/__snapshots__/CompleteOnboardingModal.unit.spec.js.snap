// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CompleteOnboardingModal > Snapshot > should match the snapshot 1`] = `
<div
  class="finish-onboarding-modal"
  data-test-id="modal-dialog"
  data-v-a24dda74=""
  hidesecondarybutton=""
  modelvalue="true"
  primarybuttonprops="[object Object]"
  showactionsdivider=""
  size="sm"
>
  
  <section
    class="finish-onboarding-modal__content"
    data-v-a24dda74=""
  >
    <h1
      class="finish-onboarding-modal__content__title"
      data-v-a24dda74=""
    >
      onboarding.complete_modal.title
    </h1>
    <p
      class="finish-onboarding-modal__content__info"
      data-v-a24dda74=""
    >
      onboarding.complete_modal.text
    </p>
  </section>
  
  
  
</div>
`;
